
rule arriba:
  input: R=lambda wildcards: FQ[wildcards.sample]
  output:
    "{subject}/{TIME}/{sample}/fusion/arriba-fusion.txt"
  envmodules:
    config["STAR-Arriba"],
    config['samtools_new'],
    config['Arriba'],
    config['version_R']
  params:
    rulename = "arriba",
    batch    = config[config['host']]['job_STAR_TPM'],
    star_gencode_ref  = config['star_2.7.6a_arriba'],
    home     = WORK_DIR,
    ref      = config['reference'],
    gtf      = config['UCSC_star'],
    blcklst  = config['Arriba_blacklist'],
    draw     = config['draw_fusions'],
    cyto     = config['cytobands'],
    prot     = config['proteinDomains'],

  shell: """

    cd ${{LOCAL}}/
    if [ -d "{params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out" ]; then rm -Rf {params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out; fi
    mkdir -p {params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out

    STAR \
      --runThreadN $SLURM_CPUS_PER_TASK \
      --genomeDir {params.star_gencode_ref} \
      --readFilesIn {input.R[0]} {input.R[1]} \
      --readFilesCommand zcat \
      --outSAMtype BAM Unsorted \
      --outSAMunmapped Within \
      --outBAMcompression 0 \
      --outFilterMultimapNmax 50 \
      --peOverlapNbasesMin 10 \
      --alignSplicedMateMapLminOverLmate 0.5 \
      --alignSJstitchMismatchNmax 5 -1 5 5 \
      --chimSegmentMin 10 \
      --chimOutType WithinBAM HardClip \
      --chimJunctionOverhangMin 10 \
      --chimScoreDropMax 30 \
      --chimScoreJunctionNonGTAG 0 \
      --chimScoreSeparation 1 \
      --chimSegmentReadGapMax 3 \
      --chimMultimapNmax 50 \
      --outFileNamePrefix {wildcards.sample}.arriba. &&
    arriba \
      -x {wildcards.sample}.arriba.Aligned.out.bam \
      -o {params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out/fusions.tsv \
      -O {params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out/fusions.discarded.tsv \
      -a {params.ref} \
      -g {params.gtf} \
      -b {params.blcklst} \
      -p {params.prot} &&
    cp {params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out/fusions.tsv {params.home}/{output}

    # Visualizations, plots all fusions predicted by Arriba
    samtools sort -o {wildcards.sample}.coordSort.arriba.Aligned.out.bam {wildcards.sample}.arriba.Aligned.out.bam
    samtools index {wildcards.sample}.coordSort.arriba.Aligned.out.bam
    {params.draw} \
      --fusions={params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out/fusions.tsv \
      --alignments={wildcards.sample}.coordSort.arriba.Aligned.out.bam \
      --output={params.home}/{wildcards.subject}/{TIME}/{wildcards.sample}/arriba_out/fusions.pdf \
      --annotation={params.gtf} \
      --cytobands={params.cyto} \
      --proteinDomains={params.prot}

    """


