# vim: syntax=python tabstop=4 expandtab
# coding: utf-8
rule HAPCALLER:
	input: 	
		bam="{subject}/{TIME}/{sample}/{sample}.bwa.final.bam",
		bai="{subject}/{TIME}/{sample}/{sample}.bwa.final.bam.bai",
		ref=config["reference"],
		dbsnp=config["dbsnp"],
		interval=lambda wildcards: config['target_intervals'][config['sample_captures'][wildcards.sample]].replace("target", "targetbp"),
	output:
		vcf="{subject}/{TIME}/{sample}/calls/{sample}.HC_DNASeq.raw.vcf"
	envmodules: config["GATK-HC"]
	params:
		rulename = "HAPCALLER",
		batch    = config[config['host']]["job_HC"]
	shell: """
	#######################
    GATK -m ${{MEM}}g HaplotypeCaller -R {input.ref} -I {input.bam} -L {input.interval} -o {output.vcf} --dbsnp {input.dbsnp} -mbq 20 -mmq 30 -nct ${{THREADS}}
#    gatk --java-options -Xmx${{MEM}}g HaplotypeCaller -R {input.ref} -I {input.bam} -L {input.interval} -O {output.vcf} --dbsnp {input.dbsnp} -mbq 20  --native-pair-hmm-threads ${{THREADS}}
	#######################
	"""
