# vim: syntax=python tabstop=4 expandtab
# coding: utf-8
if 'methylseq' in config:
	for subject in config['methylseq'].keys():
		if subject not in PATIENTS:
			PATIENTS.append(subject)
		for library in config['methylseq'][subject]:
			TARGET += [subject+"/"+TIME+"/"+library+"/qc/fastqc/"+library+"_R2_fastqc.html"]
			TARGET += [subject+"/"+TIME+"/"+library+"/"+library+"_pe.deduplicated.CpG_report.txt.gz"]
			TARGET += [subject+"/"+TIME+"/"+library+"/"+library+"_PE_report.html"]
	for subject in config['methylseq']:
		SUBJECT_TO_SAMPLE[subject] = expand("{sample}", sample = config['methylseq'][subject])
############
#       Bismark
############
rule Bismark:
	input:  R1="{base}/{TIME}/FQ/{sample}_R1.fastq.gz",
            R2 = "{base}/{TIME}/FQ/{sample}_R2.fastq.gz",
		    ref=config["BismarkIndex"]
	output:
		"{base}/{TIME}/{sample}/{sample}_pe.bam",
	envmodules:
		config["bismark"],
		config["samtools"],
		config["trimgalore"],
		config["bowtie"],
		config["fastqc"],
		config["cutadapt"]
	params:
		rulename  = "Bismark",
		platform  = config["platform"],
		batch     = config[config['host']]["job_bismark"]
	shell: """
	#######################
	trim_galore --paired --output_dir {wildcards.base}/{TIME}/{wildcards.sample}/ --trim1 {input.R1} {input.R2} 
	address=`basename {input.R1} _R1.fastq.gz`
	bismark -p ${{THREADS}} {input.ref} -1 {wildcards.base}/{TIME}/{wildcards.sample}/${{address}}_R1_val_1.fq.gz -2 {wildcards.base}/{TIME}/{wildcards.sample}/${{address}}_R2_val_2.fq.gz --output_dir {wildcards.base}/{TIME}/{wildcards.sample}/ --basename {wildcards.sample} --temp_dir ${{LOCAL}} --nucleotide_coverage 
	#######################
	"""
############
#       Bismark Deduplicate
############
rule BisDedup:
	input:
		"{base}/{TIME}/{sample}/{sample}_pe.bam"
	output:
		"{base}/{TIME}/{sample}/{sample}_pe.deduplicated.bam"
	envmodules:
		config["bismark"],
		config["samtools"],
		config["bowtie"],
	params:
		rulename  = "BisDedup",
		batch     = config[config['host']]["job_bismark"]
	shell: """
	#######################
	deduplicate_bismark -p {input} --bam
	#######################
	"""
############
#       Bismark Methylation Extractor
############
rule BismarkMethExt:
	input:
		bam="{base}/{TIME}/{sample}/{sample}_pe.deduplicated.bam",
		ref=config["BismarkIndex"]
	output:
		"{base}/{TIME}/{sample}/{sample}_pe.deduplicated.CpG_report.txt.gz",
	envmodules:
		config["bismark"],
		config["samtools"],
		config["bowtie"]
	params:
		rulename  = "BismarkMethExt",
		batch     = config[config['host']]["job_bismark"]
	shell: """
	#######################
	bismark_methylation_extractor --gzip {input.bam} --ample_memory  --multicore ${{THREADS}} --output {wildcards.base}/{TIME}/{wildcards.sample}/ --genome_folder {input.ref} --bedGraph --cytosine_report --report --no_overlap 
	#bismark_methylation_extractor {input.bam} --genome_folder {input.ref} --ample_memory  --multicore ${{THREADS}} --report --merge_non_CpG --output {wildcards.base}/{TIME}/{wildcards.sample}/ --comprehensive 
	#######################
	"""
############
#	Bismark Nucleotide Coverage
############
rule bam2nuc:
	input: "{base}/{TIME}/{sample}/{sample}_pe.bam"
	output: "{base}/{TIME}/{sample}/{sample}.nucleotide_stats.txt"
	envmodules:
		config["bismark"],
		config["samtools"],
	params:
		rulename  = "bam2nuc",
		batch     = config[config['host']]["job_bismark"],
		ref=config["BismarkIndex"]
	shell: """
	#######################
	bam2nuc --dir {wildcards.base}/{TIME}/{wildcards.sample}/ --genome_folder {params.ref} {input}
	#######################
	"""
############
#       Bismark HTML Report
############
rule BisReport: 
	input:
		"{base}/{TIME}/{sample}/{sample}_pe.deduplicated.CpG_report.txt.gz",
		#"{base}/{TIME}/{sample}/{sample}.nucleotide_stats.txt"
	output:
		"{base}/{TIME}/{sample}/{sample}_PE_report.html"
	envmodules:
		config["bismark"],
		config["samtools"],
	params:
		rulename  = "BisReport",
		batch     = config[config['host']]["job_default"]
	shell: """
	#######################
	bismark2report --dir {wildcards.base}/{TIME}/{wildcards.sample}/ --alignment_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_PE_report.txt --dedup_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplication_report.txt --splitting_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplicated_splitting_report.txt --mbias_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplicated.M-bias.txt
	#bismark2report --dir {wildcards.base}/{TIME}/{wildcards.sample}/ --alignment_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_PE_report.txt --dedup_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplication_report.txt --splitting_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplicated_splitting_report.txt --mbias_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}_pe.deduplicated.M-bias.txt --nucleotide_report {wildcards.base}/{TIME}/{wildcards.sample}/{wildcards.sample}
	#######################
	"""
